File: system-design/oop-patterns/02-observer-pattern/full/java/Main.java
import weatherstation.WeatherStation;
import weatherstation.PhoneDisplay;
import weatherstation.WindowDisplay;

public class Main {
    public static void main(String[] args) {
        WeatherStation station = new WeatherStation();

        PhoneDisplay phone = new PhoneDisplay(station);
        WindowDisplay window = new WindowDisplay(station);
        station.add(phone);
        station.add(window);

        station.setTemperature(72);
        station.setTemperature(68);
    }
}

File: system-design/oop-patterns/02-observer-pattern/full/cpp/main.cpp
#include "weather-station/WeatherStation.h"
#include "weather-station/PhoneDisplay.h"
#include "weather-station/WindowDisplay.h"

int main() {
    WeatherStation station;

    PhoneDisplay phone(&station);
    WindowDisplay window(&station);
    station.add(&phone);
    station.add(&window);

    station.setTemperature(72);
    station.setTemperature(68);

    return 0;
}
